src/main/java/learn/house/ui/View.java

```diff
@@ -0,0 +1,355 @@
+    private String formatHost(int index, Host host) {
+        return String.format(hostTemplate,
+                index++, host.getStandardRate().toString(), host.getWeekendRate().toString(), host.getLastName(),
+                host.getPhone(), host.getEmail(), host.getAddress(), host.getCity(), host.getState(), host.getPostalCode());
+    }
+
+    private String formatGuest(int index, Guest guest) {
+        return String.format(guestTemplate,
+                index++, guest.getLastName(), guest.getFirstName(),
+                guest.getPhone(), guest.getEmail(), guest.getState());
+    }
+
+    private String formatReservation (int index, Reservation reservation) {
+        return String.format(reservationTemplate,
+                index++,
+                reservation.getStartDate(),
+                reservation.getEndDate(),
+                reservation.getGuestID(),
+                reservation.getTotal()
+        );
+    }
+
+
+
+    public Host selectHost(List<Host> hosts) {
+        if (hosts.size() == 0) {
+            io.println("No hosts found");
+            return null;
+        }
+
+        int index = 1;
+        for (Host host : hosts.stream().limit(25).collect(Collectors.toList())) {
+            io.println(formatHost(index++, host));
+        }
+
+        if (hosts.size() > 25) {
+            io.println("More than 25 hosts found. Showing first 25. Please refine your search.");
+        }
+
+        io.println("0: Exit");
+        String message = String.format("Select a host by their index [1-%s]: ", index-1);
+
+        index = io.readInt(message, 0, index-1);
+        if (index <= 0) {
+            return null;
+        }
+        return hosts.get(index - 1);
+    }
+
+    public Host getHostCriteria() {
+        io.println("Enter criteria to filter Hosts by equivalence (empty fields will be ignored): ");
+        String id = io.readString("HostID: ");
+        String lastName = io.readString("Last Name: ");
+        String email = io.readEmailOptional("Email: ");
+        String phone = io.readString("Phone #: ");
+        String address = io.readString("address: ");
+        String city = io.readString("City: ");
+        States state = io.readStateOptional("State (two letters): ");
+        String postalCode = io.readString("Postal Code: ");
+        BigDecimal standardRate = io.readBigDecimalOptional("Standard Rate (no symbols): ");
+        BigDecimal weekendRate = io.readBigDecimalOptional("Weekend Rate (no symbols: ");
+
+        return new Host(
+               id.equals("") ? null : id,
+               lastName.equals("") ? null : lastName,
+               email,
+               phone.equals("") ? null : phone,
+               address.equals("") ? null : address,
+               city.equals("") ? null : city,
+               state,
+               postalCode.equals("") ? null : postalCode,
+               standardRate,
+               weekendRate);
+    }
```

Comment:
From the UI/UX perspective, this is cumbersome in a console app. Not a requirement, but a better experience would have been to prompt the user to select which criteria to search against, i.e. a Switch.